src/ProjectManagement/Strings.resx

```diff
@@ -234,6 +234,9 @@
   <data name="RemovingPackageFromMSBuildProject" xml:space="preserve">
     <value>Removing package '{0}' from project '{1}'</value>
   </data>
+  <data name="SolutionDirectoryShouldBeANonNullFullPath" xml:space="preserve">
+    <value>Solution directory must be a non-null full path</value>
```

Comment:
Done. First checking for nullity. Throwing an ArgNull on solutionDirectory if null. If not, and if the path is not rooted, saying that the "SolutionDirectory '{0}' must be a rooted path."